x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/elastic/ElasticInferenceServiceSparseEmbeddingsRequestEntity.java

```diff
@@ -34,8 +40,15 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
 
         builder.endArray();
+
+        // optional field
+        if ((usageContext == ElasticInferenceServiceUsageContext.UNSPECIFIED) == false) {
```

Comment:
```suggestion
        if (usageContext != ElasticInferenceServiceUsageContext.UNSPECIFIED) {
```
Doesn't this work too?

Reply:
Unfortunately not in the Elasticsearch codebase as checkstyle forces you to use `condition == false` instead of `!condition`.

Reply:
TIL